app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -94,6 +95,18 @@ def fetch_server_entities
         end
       end
 
+      def fetch_availability
+        metrics_ids = {}
+        @data[:middleware_deployments].each do |deployment|
+          metric_id = @ems.build_metric_id('A', deployment, 'Deployment Status~Deployment Status')
+          metrics_ids[metric_id] = deployment
+        end
+        availabilities = @ems.metrics_client.avail.raw_data(metrics_ids.keys, :limit => 1)
+        availabilities.each do |availability|
+          metrics_ids[availability['id']][:status] = process_availability(availability['data'].first)
```

Comment:
metrics_ids contains already parsed data. availability is coming directly from the server. That's the reason it uses `id` instead of `:id`